samwise.js

```diff
@@ -20,11 +20,11 @@
     // handle step-1.
```

Comment:
SCRIPT-The way you've refactored this complex method into simpler ones is great. It makes the code much more maintainable.